### PR TITLE
Always use x64 ilasm when building on Windows

### PR DESCRIPTION
**Pull request by mellinoe**
@weshaggard @MattGal

### CODE REVIEW

File: external/ilasm/ilasm.depproj [hunk 1]
```diff
@@ -2,6 +2,8 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
+    <!-- Always restore the win7-x64 assets on Windows. These are build-only assets, so they should match the environment of the build. -->
+    <NugetRuntimeIdentifier Condition="'$(RunningOnUnix)' != 'true'">win7-x64</NugetRuntimeIdentifier>
```
Comment (weshaggard) [inline external/ilasm/ilasm.depproj]: What about the unix builds? If feels like we just always need to override the ArchGroup to be x64 for our tools packages but the Identifier should still match.

Comment (weshaggard) [inline external/ilasm/ilasm.depproj]: https://github.com/dotnet/corefx/blob/master/external/dir.props#L5

I think we just want to away set:

`<NugetRuntimeIdentifier>$(RuntimeOS)-x64</NugetRuntimeIdentifier>`

Comment (mellinoe) [inline external/ilasm/ilasm.depproj]: Sounds good. We essentially want the "runtime RID" which matches the build environment. If we could have something like `$(RuntimeOS)-$(RuntimeArch)`, that would be good.

**Comment by mellinoe** [comment]
I also made sure that the ilasm project is marked `<IsRuntimeAssembly>false</IsRuntimeAssembly>` so that it isn't being copied into the runtime folder. We don't want that anyways, and now the architectures aren't guaranteed to match.

File: external/ilasm/ilasm.depproj [hunk 2]
```diff
@@ -2,7 +2,10 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
+    <!-- Always restore the win7-x64 assets on Windows. These are build-only assets, so they should match the environment of the build. -->
+    <NugetRuntimeIdentifier Condition="'$(RunningOnUnix)' != 'true'">$(RuntimeOS)-x64</NugetRuntimeIdentifier>
```
Comment (weshaggard) [inline external/ilasm/ilasm.depproj]: I think you want to remove the "RunningOnUnix" condition.

**Pull request merged**